lib/usd/translators/shading/mtlxFileTextureWriter.cpp

```diff
@@ -148,45 +150,67 @@ MtlxUsd_FileWriter::MtlxUsd_FileWriter(
+
+        // We expose the primvar reader varnameStr attribute to the material to allow
+        // easy specialization based on UV mappings to geometries:
+        UsdPrim          materialPrim = primvarReaderSchema.GetPrim().GetParent();
+        UsdShadeMaterial materialSchema(materialPrim);
+        while (!materialSchema && materialPrim) {
+            UsdShadeNodeGraph intermediateNodeGraph(materialPrim);
+            if (intermediateNodeGraph) {
+                UsdShadeInput intermediateInput
+                    = intermediateNodeGraph.CreateInput(inputName, SdfValueTypeNames->String);
+                varnameInput.ConnectToSource(intermediateInput);
+                varnameInput = intermediateInput;
+            }
 
-    UsdShadeOutput primvarReaderOutput
-        = primvarReaderSchema.CreateOutput(TrMtlxTokens->out, SdfValueTypeNames->Float2);
+            materialPrim = materialPrim.GetParent();
+            materialSchema = UsdShadeMaterial(materialPrim);
+        }
 
-    // TODO: Handle UV SRT with a ND_place2d_vector2 node.
+        if (materialSchema) {
+            UsdShadeInput materialInput
+                = materialSchema.CreateInput(inputName, SdfValueTypeNames->String);
+            materialInput.Set(UsdUtilsGetPrimaryUVSetName().GetString());
+            varnameInput.ConnectToSource(materialInput);
+        } else {
+            varnameInput.Set(UsdUtilsGetPrimaryUVSetName());
+        }
+
+        UsdShadeOutput primvarReaderOutput
+            = primvarReaderSchema.CreateOutput(TrMtlxTokens->out, SdfValueTypeNames->Float2);
+
+        // TODO: Handle UV SRT with a ND_place2d_vector2 node. Make sure the name derives from the
+        //       place2dTexture node if there is one (see usdFileTextureWriter for details)
+
+        // Connect the output of the primvar reader to the texture coordinate
+        // input of the UV texture.
+        texSchema.CreateInput(TrMtlxTokens->texcoord, SdfValueTypeNames->Float2)
+            .ConnectToSource(primvarReaderOutput);
+    } else {
+        // Re-using an existing primvar reader:
+        UsdShadeShader primvarReaderShaderSchema(GetUsdStage()->GetPrimAtPath(primvarReaderPath));
+        UsdShadeOutput primvarReaderOutput = primvarReaderShaderSchema.GetOutput(TrMtlxTokens->out);
 
-    // Connect the output of the primvar reader to the texture coordinate
-    // input of the UV texture.
-    texSchema.CreateInput(TrMtlxTokens->texcoord, SdfValueTypeNames->Float2)
-        .ConnectToSource(primvarReaderOutput);
+        // TODO: Handle UV SRT with a ND_place2d_vector2 node. Make sure the name derives from the
+        //       place2dTexture node if there is one (see usdFileTextureWriter for details)
+
+        // Connect the output of the primvar reader to the texture coordinate
+        // input of the UV texture.
+        texSchema.CreateInput(TrMtlxTokens->texcoord, SdfValueTypeNames->Float2)
+            .ConnectToSource(primvarReaderOutput);
```

Comment:
Line 206-212 could be taken out of the if/else block instead of having a duplication of code.